Frontend/Frontend.Client/Models/ChannelManager.cs

```diff
@@ -0,0 +1,104 @@
+﻿using LSMP;
+using static Frontend.Client.Models.ChannelManager;
+
+namespace Frontend.Client.Models
+{
+    public static class ChannelManager
+    {
+        public class ChannelData
+        {
+            public List<UserMessage> MessageHistory { get; set; } = new List<UserMessage>();
+            public List<string> UserList { get; set; } = new List<string>();
+        }
+
+        public static Dictionary<string, ChannelData> channels = new Dictionary<string, ChannelData>
+        {
+            {"general-chat", new ChannelData() }
+        };
+
+        public static string CurrentChannel { get; set; } = "general-chat";
+
+        public static event Action? OnStateChange;
+
+        public static string CreateDmChannel(string targetUser)
+        {
+            var channelData = new ChannelData
```

Comment:
Should first check if it exists in the channels list and returns the channel name if it exists otherwise continues to create it

Reply:
added a "ContainsKey" check